### ISSUE TITLE
[FEATURE] Implement Responsive View Logic for Bottom Sheet and Modal

### ISSUE THREAD

**Issue by manideepk90** [juspay/hyperswitch#6231]
# Description

The [Hyperswitch SDK](https://docs.hyperswitch.io/hyperswitch-cloud/integration-guide) aims to dynamically switch between a bottom sheet for smaller screens, like mobile devices, and a modal view for larger screens, such as tablets and desktops. This approach ensures optimal use of screen real estate and allows for seamless interactions regardless of the device being used, which is crucial for a smooth payment process.

Currently, when users open the payment sheet in their app using HyperSwitch, it appears as a bottom sheet occupying the full width, even on larger screens. To improve this experience, the bottom sheet should function as a modal on larger screens while maintaining its design for smaller screens. This change will make the payment process more intuitive and user-friendly, adapting to the specific needs of each device and enhancing overall usability.

### Getting started
1. Get familiar with [ReScript](https://rescript-lang.org/docs/manual/latest/introduction).
2. Check the [README.md](https://github.com/juspay/hyperswitch-client-core/blob/main/README.md) for project structure and setup instructions.
3. To setup locally, follow these steps

```  bash
#  Clone repository
cd hyperswitch-client-core
git submodule update --init --recursive
yarn install
yarn run server     # This starts the mock server
yarn run re:start   # In another terminal
```

### **Expected Behavior:**
On smaller screens (mobile), a bottom sheet should appear.
On larger screens (tablet/desktop), the modal view should appear.
Transitions between the views should be smooth and without glitches when resizing the window or rotating the device.
<img width="600" height="360" alt="Expected behavior" src="https://github.com/user-attachments/assets/7d5e7835-c02e-46a1-be84-c7fe711ab406">

### **Actual Behavior:**
The current implementation only supports a single view (either bottom sheet or modal) without dynamically adjusting to screen size.
<img width="600" height="360" alt="Actual Behavior" src="https://github.com/user-attachments/assets/98299ecc-6ea8-4efa-a31e-aa17620d069d">

## **Tasks:**
Screen Size Detection :

Implement logic to detect screen size or device type at runtime.

### **Switching Logic:**
Develop a mechanism to toggle between the bottom sheet for smaller screens and modal for larger screens.

### **Seamless Transition:**
Ensure the transition between the two view types is smooth, especially when resizing the window or switching between devices.

### **Testing:**
Test the implementation on various screen sizes and devices (mobile, tablet, desktop) to ensure that the views transition seamlessly.

## Contribution Guidelines:

- Fork the repository and create a new branch for your work.
- Write clean, well-documented code with clear commit messages.
- Make sure to follow our coding standards and contribution guidelines.

If you have any questions or need help getting started, feel free to ask in the comments!

**Comment by adiityack** [comment]
Can you please assign this issue to me? This will be my first contribution to open source, and I’m excited to work on it. I will need lots of help along the way, but I’m eager to learn!

**Comment by gorakhnathy7** [comment]
Sure @adiityack Happy to support on this journey.

**Comment by adiityack** [comment]
hey, I have done with ( Set up a development environment using Docker Compose ) what should I do next ? how to approach to solve this issue.

**Comment by manideepk90** [comment]
you can start running application locally and then you need to go through the bottom sheet view. 
convert it as modal for large screens.
make sure you won't break logic and add fade animation to it while opening and closing. 
familiarize with rescript yourself. follow our [community](https://hyperswitch-io.slack.com/archives/C04JXRZ338A) for support

**Comment by adiityack** [comment]
Thank you for the guidance! I'm seeking help from the community as well and will try my best to resolve this. Since I'm a beginner with open-source development, could you please let me know the deadline for this task, or if there's any flexibility until I can figure everything out?

**Comment by adiityack** [comment]
cd hyperswitch-client-core
git submodule update --init --recursive
yarn install
yarn run server     # This starts the mock server
yarn run re:start  I have folk the repo and cloned it after running all this command I can not run the local host 
<img width="152" alt="Screenshot 2024-10-08 at 1 28 58 PM" src="https://github.com/user-attachments/assets/bc1621c2-3a90-44b4-951b-a6d661c042b2">
this is what I can see on my screen how to fix this?

**Comment by manideepk90** [comment]
First, you need to add your publishable key and API in the .env file. Run `yarn start` to launch the Metro server, then press `a` to open Android or `i` for iOS. To work on the web version, run `yarn web` and `yarn server` simultaneously. Then, open the file DemoAppIndex.html located at ./reactNativeDemo/DemoAppIndex.html in your browser. You are now ready to begin working on the web version.

**Comment by adiityack** [comment]
I have just created HYPERSWITCH_API_KEY and HYPERSWITCH_PUBLISHABLE_KEY and have putted in the .en file. and have used the command` yarn web` and `yarn server` and still facing the same issue. Is I have to paste that on .en file? and what about other section secret key,sdn,code push android ,code push iOS.  were I am getting wrong please help.

**Comment by manideepk90** [comment]
Yes, rename the file to `.env` or create a file `.env` . When submitting, ensure that you do not commit the `.env` file and add dummy connectors in the Hyperswitch dashboard to view the payment options.

**Comment by adiityack** [comment]
I have successfully completed the setup of the development environment and am now working on the task assigned to me.

**Comment by gorakhnathy7** [comment]
Hey @adiityack 


Kindly let us know, if you're still working on the issue? If you have any further questions or need assistance, feel free to reach out in the [community](https://join.slack.com/t/hyperswitch-io/shared_invite/zt-2jqxmpsbm-WXUENx022HjNEy~Ark7Orw).
